### PR TITLE
Frontend/bottom_sheet_layout_Changes

### PR DESCRIPTION
**Pull request by KiranBKL**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/android-native/app/src/user/nammaYatri/assets/juspay/juspay_assets.json [hunk 1]
```diff
@@ -203,6 +205,7 @@
       "ny_ic_icon" : true,
       "ny_ic_info" : true,
       "ny_ic_info_blue" : true,
+      "ny_ic_info_blue_large" : true,
```
Comment (ashishsingh101) [inline Frontend/android-native/app/src/user/nammaYatri/assets/juspay/juspay_assets.json:208]: we already have this icon with name `ny_ic_info_blue` please use that one

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs [hunk 2]
```diff
@@ -1077,8 +1077,37 @@ requestInfoCardConfig _ = let
...
+      text = getString YOU_WILL_PAY_FOR_EVERY_MINUTE,
+      visibility = VISIBLE,
+      padding = PaddingLeft 16
+    }
+  , imageConfig {
+      imageUrl = "ny_ic_waiter," <> (getAssetStoreLink FunctionCall) <> "ny_ic_waiter.png",
+      height = V 130,
+      width = V 130,
+      padding = Padding 0 4 1 0
+    }
```
Comment (ashishsingh101) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs:1101]: can you check as per latest figma if this is required ?

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs [hunk 3]
```diff
@@ -1315,9 +1325,25 @@ topLeftIconView state push =
           [ height WRAP_CONTENT
           , weight 1.0
           ][]
-        , referralView push state
-        , if ((getValueFromConfig "showDashboard") == "false") || (isPreviousVersion (getValueToLocalStore VERSION_NAME) (if os == "IOS" then "1.2.5" else "1.2.1")) then emptyTextView state else liveStatsDashboardView push state
+       , sosView push state
+    --    , if ((getValueFromConfig "showDashboard") == "false") || (isPreviousVersion (getValueToLocalStore VERSION_NAME) (if os == "IOS" then "1.2.5" else "1.2.1")) then emptyTextView state else liveStatsDashboardView push state
```
Comment (ashishsingh101) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:1329]: add `TODO` why we commented this line and why its required ?

File: Frontend/ui-customer/src/Screens/Types.purs [hunk 4]
```diff
@@ -583,6 +583,8 @@ type HomeScreenStateData =
   , ratingViewState :: RatingViewState
   , config :: AppConfig
   , logField :: Object Foreign
+  --,activeRideWaitTimeInfo :: Boolean
```
Comment (ashishsingh101) [inline Frontend/ui-customer/src/Screens/Types.purs:586]: remove commented line

**Review by ashishsingh101 — CHANGES_REQUESTED** [review]

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs (hunk 2 above)
Comment (ashishsingh101) [reply Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs:1101]: remove image also if its not required

File: Frontend/ui-customer/src/Components/DriverInfoCard/View.purs [hunk 5]
```diff
@@ -586,7 +702,15 @@ waitTimeView push state =
...
-      , color Color.black800
+      , color (case split (Pattern ":") state.data.waitingTime of
+             [minutes, _] ->
+               if minutes == "00 " 
+               then Color.green900  
+               else if minutes == "01 " || minutes == "02 "
+                    then Color.yellow900 
+                    else Color.orange800   
+             _ -> Color.black 
+          )
```
[the remark this answers is not in the document]
Comment (KiranBKL) [reply Frontend/ui-customer/src/Components/DriverInfoCard/View.purs:712]: Yeah,i checked that there three colors

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs [hunk 6]
```diff
@@ -930,6 +930,56 @@ callSupportConfig state = let
...
+    , color = state.data.config.primaryBackground
+    , width = MATCH_PARENT
+    , margin = (Margin 0 10 0 0)
+    }
+  , option1 {
+     text =  getString SHARE_THIS_RIDE
+    , color = state.data.config.primaryTextColor
+    , strokeColor = state.data.config.primaryBackground
+    , background = state.data.config.primaryBackground
+    , margin = (Margin 0 0 0 0)
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs:977]: remove extra brackets

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs [hunk 7]
```diff
@@ -894,7 +901,7 @@ eval (UpdateSource lat lng name) state = do
 
 eval (HideLiveDashboard val) state = continue state {props {showLiveDashboard =false}}
 
-eval LiveDashboardAction state = do
+eval (DriverInfoCardActionController (DriverInfoCardController.LiveDashboardAction)) state = do
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs:904]: normal dashboard flow will not work if we do like this. create separate eval of this and call `LiveDashboardAction`inside it.

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]

**Comment by Vignesh-772** [comment]
Screenshot_1693031712
able to see unwanted margin in top. check and remove that also.

File: Frontend/android-native/app/src/user/jatriSaathi/res/drawable/ny_ic_waiter.png
[the remark this answers is not in the document]
Comment (KiranBKL) [reply Frontend/android-native/app/src/user/jatriSaathi/res/drawable/ny_ic_waiter.png:1]: ha Size has been reduced

File: Frontend/android-native/app/src/user/jatriSaathi/res/drawable/ic_message_filled.png
[the remark this answers is not in the document]
Comment (KiranBKL) [reply Frontend/android-native/app/src/user/jatriSaathi/res/drawable/ic_message_filled.png:1]: Image is already there actually

File: Frontend/ui-common/src/Components/PopUpModal/Controller.purs [hunk 8]
```diff
@@ -138,6 +139,15 @@ config = {
   , editTextVisibility : GONE
   , tipLayoutMargin : (Margin 0 0 0 0)
   , padding : (Padding 0 0 0 0)
+  , title : {
```
[the remark this answers is not in the document]
Comment (KiranBKL) [reply Frontend/ui-common/src/Components/PopUpModal/Controller.purs:142]: yeah sure

File: Frontend/ui-customer/src/Components/DriverInfoCard/View.purs [hunk 9]
```diff
@@ -695,9 +840,9 @@ cancelRideLayout push state =
   ][ textView (
      [ width WRAP_CONTENT
```
[the remark this answers is not in the document]
Comment (KiranBKL) [reply Frontend/ui-customer/src/Components/DriverInfoCard/View.purs:841]: yeah

File: Frontend/android-native/app/src/user/nammaYatri/res/drawable/ic_message_filled_notified.png
Comment (witcher-shailesh) [inline Frontend/android-native/app/src/user/nammaYatri/res/drawable/ic_message_filled_notified.png:1]: fix the image resolution

File: Frontend/android-native/app/src/user/yatri/res/drawable/ic_message_filled_notified.png
Comment (witcher-shailesh) [inline Frontend/android-native/app/src/user/yatri/res/drawable/ic_message_filled_notified.png:1]: check resolution

File: Frontend/android-native/app/src/user/jatriSaathi/res/drawable/ic_message_filled.png
[the remark this answers is not in the document]
Comment (Vignesh-772) [reply Frontend/android-native/app/src/user/jatriSaathi/res/drawable/ic_message_filled.png:1]: it says newly added

File: Frontend/android-native/app/src/user/jatriSaathi/res/drawable/ic_message_filled_notified.png
[the remark this answers is not in the document]
Comment (Vignesh-772) [reply Frontend/android-native/app/src/user/jatriSaathi/res/drawable/ic_message_filled_notified.png:1]: this is not resolved

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]

File: Frontend/ui-customer/src/Components/DriverInfoCard/View.purs [hunk 10]
```diff
@@ -586,7 +696,15 @@ waitTimeView push state =
...
-      , color Color.black800
+      , color (case split (Pattern ":") state.data.waitingTime of
+             [minutes, _] ->
+               if minutes == "00 " 
+               then Color.green900  
+               else if minutes == "01 " || minutes == "02 "
+                    then Color.yellow900 
+                    else Color.orange800   
+             _ -> Color.black 
+          )
```
Comment (ashishsingh101) [inline Frontend/ui-customer/src/Components/DriverInfoCard/View.purs:707]: don't do like this, create a function in `common.js` which will have arguments date and format, there you can send `mm` to get minutes

**Review by ashishsingh101 — CHANGES_REQUESTED** [review]

File: Frontend/ui-customer/src/Components/DriverInfoCard/View.purs [hunk 11]
```diff
@@ -731,39 +876,23 @@ contactView push state =
                 , color Color.black800
```
[the remark this answers is not in the document]
Comment (KiranBKL) [reply Frontend/ui-customer/src/Components/DriverInfoCard/View.purs:871]: We do not need translation when driver is away

File: Frontend/ui-customer/src/Components/DriverInfoCard/View.purs (hunk 10 above)
Comment (KiranBKL) [reply Frontend/ui-customer/src/Components/DriverInfoCard/View.purs:707]: Dont required ,we r already getting waiting time state data variable

**Pull request closed without merging**